recipes/tinycthreadpool/all/conanfile.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from conan import ConanFile
+from conans import CMake, tools
+from conan.tools.files import get
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.53.0"
+
+
+class TinyCThreadPoolConan(ConanFile):
+    name = "tinycthreadpool"
+    description = "Portable C thread pool"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/bennyhuo/tinycthreadpool"
+    topics = ("thread-pool", "threading", "pure-c")
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeDeps"
+    
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(source_folder="src")
+        cmake.build()
+
+    def package(self):
+        self.copy("*.h", dst="include", src="src")
+        self.copy("*.lib", dst="lib", keep_path=False)
+        self.copy("*.dll", dst="bin", keep_path=False)
+        self.copy("*.dylib*", dst="lib", keep_path=False)
+        self.copy("*.so", dst="lib", keep_path=False)
+        self.copy("*.a", dst="lib", keep_path=False)
+        self.copy("LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["tinycthreadpool"]
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs = ["pthread"]
+
+    def validate(self):
+        if self.settings.compiler == "Visual Studio":
+            if Version(self.settings.compiler.version) < "16":
+                raise ConanInvalidConfiguration(f"TinyCThreadPool doesn't support MSVC version: {self.settings.compiler.version}")
+            if self.settings.compiler.runtime in ['MTd', 'MT']:
+                raise ConanInvalidConfiguration(f"TinyCThreadPool doesn't support this runtime {self.settings.compiler.runtime}")
```

Comment:
```suggestion
```
Let's move it to the correct order, https://docs.conan.io/en/latest/reference/commands/creator/create.html#methods-execution-order